gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/EncounterMapper.java

```diff
@@ -0,0 +1,327 @@
+
+        return encounter;
+    }
+
+    private List<CodeableConcept> getEncounterType(CD code) {
+        return List.of(codeableConceptMapper.mapToCodeableConcept(code));
+    }
+
+    private Meta getEncounterMeta() {
+        return new Meta().addProfile(ENCOUNTER_META_PROFILE);
+    }
+
+    private List<Identifier> getEncounterIdentifier(String id) {
+        Identifier identifier = new Identifier()
+            .setSystem(IDENTIFIER_SYSTEM) // TODO: concatenate source practice org id to URL (NIAD-2021)
+            .setValue(id);
+        return List.of(identifier);
+    }
+
+    private Period getPeriod(IVLTS effectiveTime, TS availabilityTime) {
+        Period period = new Period();
+        var center = getTSStringValue(effectiveTime.getCenter());
+        var low = getIVXBTSStringValue(effectiveTime.getLow());
+        var high = getIVXBTSStringValue(effectiveTime.getHigh());
+        var availabilityTimeValue = availabilityTime.getValue();
+
+        if (validValue(center)) {
+            return period.setStartElement(DateFormatUtil.parseToDateTimeType(center));
+        } else if (validValue(low) && validValue(high)) {
+            return period.setStartElement(DateFormatUtil.parseToDateTimeType(low))
+                .setEndElement(DateFormatUtil.parseToDateTimeType(high));
+        } else if (validValue(low) && !validValue(high)) {
+            return period.setStartElement(DateFormatUtil.parseToDateTimeType(low));
+        } else if (!validValue(low) && validValue(high)) {
+            return period.setEndElement(DateFormatUtil.parseToDateTimeType(high));
+        } else if (CsNullFlavor.UNK.value().equals(center)) {
+            return null;
+        } else if (validValue(availabilityTimeValue)) {
+            return period.setStartElement(DateFormatUtil.parseToDateTimeType(availabilityTimeValue));
+        }
+
+        return null;
+    }
+
+    private String getTSStringValue(TS ts) {
+        if (ts == null) {
+            return null;
+        } else if (ts.getValue() != null) {
+            return ts.getValue();
+        } else if (ts.getNullFlavor().equals(CsNullFlavor.UNK)) {
+            return CsNullFlavor.UNK.value();
+        }
+
+        return null;
+    }
+
+    private String getIVXBTSStringValue(IVXBTS ivxbts) {
+        if (ivxbts == null) {
+            return null;
+        } else if (ivxbts.getValue() != null) {
+            return ivxbts.getValue();
+        } else if (ivxbts.getNullFlavor().equals(CsNullFlavor.UNK)) {
+            return CsNullFlavor.UNK.value();
+        }
+
+        return null;
+    }
+
+    private boolean validValue(String value) {
+        return value != null && !CsNullFlavor.UNK.value().equals(value);
+    }
+
+    private EncounterParticipantComponent getRecorder(RCMRMT030101UK04Author author) {
+        var recorder = new EncounterParticipantComponent();
+        var coding = new Coding();
+
+        coding
+            .setSystem(RECORDER_SYSTEM)
```

Comment:
minor - Coding has a constructor that takes these 3 values so we could do that instead (applies below too)